tests/deployment_tests/test/gitlabTest.js

```diff
@@ -0,0 +1,179 @@
+/* eslint no-underscore-dangle: [2, { "allow": ["__get__"] }] */
+const chai = require('chai');
+const chaiAsPromised = require('chai-as-promised');
+const rewire = require('rewire');
+const dirtyChai = require('dirty-chai');
+const { User, Project, listUsers } = require('../gitlab.js');
+const { gitlab } = require('../../../deploy/configs/gitlab/gitlab.json');
+const git = require('simple-git/promise');
+
+const getToken = rewire('../gitlab.js').__get__('getToken');
```

Comment:
There is some discussion of [sinon-vs-rewire](https://joost.vunderink.net/blog/2016/03/27/sinon-vs-rewire-when-do-i-use-which/). I am yet to wrap my head around it.

If we can get the job done with sinon alone, we shoud do so. More libraries are a mental burden in the long run.

Reply:
please see the testing choices expressed in [this issue](https://github.com/AutolabJS/autolabcli/issues/39) that are relevant to testing gitlab.js.